pullman/storageproviders/pvc/provider.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pvcprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kserve/modelmesh-runtime-adapter/internal/util"
+	"github.com/kserve/modelmesh-runtime-adapter/pullman"
+)
+
+const (
+	// config fields
+	configPVCName      = "name"
+	configPVCMountBase = "pvc_mount_base"
+
+	// defaults
+	defaultPVCMountBase = "/pvc_mounts"
+)
+
+type pvcProvider struct {
+}
+
+// pvcProvider implements StorageProvider
+var _ pullman.StorageProvider = (*pvcProvider)(nil)
+
+func (p pvcProvider) GetKey(config pullman.Config) string {
+	// hash the values of all configurations that go into creating the client
+	// no need to validate the config here
+	pvcName, _ := pullman.GetString(config, configPVCName)
+	pvcMountBase, _ := pullman.GetString(config, configPVCMountBase)
+
+	return pullman.HashStrings(pvcName, pvcMountBase)
+}
+
+func (p pvcProvider) NewRepository(config pullman.Config, log logr.Logger) (pullman.RepositoryClient, error) {
+
+	pvcName, _ := pullman.GetString(config, configPVCName)
+
+	if pvcName == "" {
+		return nil, errors.New("pvc name must be specified")
+	}
+
```

Comment:
I don't think there's any need to check config here.. it would make sense to check for the existence of the `/pvc_mounts` dir here however.

`io.Stat` is the right func to use for this, `ReadDir` is for listing dir contents.

You can do:
```go
if fi, err := os.Stat(path); err != nil || !fi.IsDir() {
    // problem
}
```

Reply:
What if the path is file? is.Stat wouldn't catch it. I believe.

Reply:
Are you saying pvcName is guaranteed to be in the config and we don't care whether it is provided or not? And, if it is not provided, we treat /pvc_mounts as the pvcName folder?

Reply:
> What if the path is file? os.Stat wouldn't catch it. I believe. If preferred, the returned FileInfo from os.Stat can be further checked with IsDir to ensure the path string is indeed a directory.

Right, I'm just suggesting to use this to verify that it exists and is a directory.

> Are you saying pvcName is guaranteed to be in the config and we don't care whether it is provided or not? And, if it is not provided, we treat /pvc_mounts as the pvcName folder?

No not saying this, I'm just saying that this isn't the right place to check for the existence of the pvc directory, since with the other change I'm suggesting, this will only be called once and then reused.